docs/advanced/integration_tests.md

```diff
@@ -0,0 +1,130 @@
+# Integration tests (advanced)
+
+There are several types of Gosoline integration tests, each of them meant to be used for a specific situation. All of them have the same thing in common: the Gosoline `suite` package is responsible for configuring, starting and running an application, then the tests cases are run.
+
+### resty.Client integration tests
+
+In the [Integration tests for your API](../getting_started/integration_tests.md) example, we saw the following test:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) Test_ToEuro/ /\n}/)
+```go
+func (s *ApiTestSuite) Test_ToEuro(_ suite.AppUnderTest, client *resty.Client) error {
+	var result float64
+
+	response, err := client.R().
+		SetResult(&result).
+		Execute(http.MethodGet, "/euro/10/GBP")
+
+	s.NoError(err)
+	s.Equal(http.StatusOK, response.StatusCode())
+	s.Equal(8.0, result)
+
+	return nil
+}
+```
+
+The _Test_ToEuro_ method makes a GET call to an endpoint, in order to receive back the euro exchange value for 10 BGP. Lastly, it checks if the received value is 8.0.
+
+In order to make this GET call _Test_ToEuro_ does not need to concern itself about the IP on which the exchange application is running, nor its port. All _Test_ToEuro_ needs to know is the URL path of that endpoint, as the _client_ object does the rest.
+
+This _client_ object is provided out-of-the box by Gosoline, whenever a test is defined like this:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) SetupSuite/ /return definer.ApiDefiner\n}/)
+```go
+func (s *ApiTestSuite) SetupSuite() []suite.Option {
+	return []suite.Option{
+		suite.WithLogLevel("info"),
+		suite.WithConfigFile("../api/config.dist.yml"),
+		suite.WithFixtures(fixtureSets),
+		suite.WithClockProvider(s.clock),
+	}
+}
+
+func (s *ApiTestSuite) SetupApiDefinitions() apiserver.Definer {
+	return definer.ApiDefiner
+}
+```
+
+Implementing _SetupApiDefinitions_ when configuring your test will inform Gosoline that an API is being tested, thus the _client_ object will contain meaningful data. 
+
+### ApiServer integration tests
+
+The _client_ object is permissive, allowing access to the request itself. A simpler way to test an API, is provided by Gosoline's _ApiServerTestCase_. These type of tests also issue http calls and compare their responses against a predefined value, but they need to follow a predefined structure. Below we can see how the _resty client_ test above would look like as an _ApiServerTestCase_:
```

Comment:
Delete the comma after "A simpler way to test an API"

Also, when do you write http and when HTTP?